great_expectations/data_context/data_context/abstract_data_context.py

```diff
@@ -11,4 +46,459 @@ class AbstractDataContext(ABC):
+            states["TRUE"] = True
+            config.BOOLEAN_STATES = states
+            config.read(config_path)
+            try:
+                if config.getboolean("anonymous_usage_statistics", "enabled") is False:
+                    # If stats are disabled, then opt out is true
+                    return True
+            except (ValueError, configparser.Error):
+                pass
+
+    def _init_stores(self, store_configs: Dict[str, dict]) -> None:
+        """Initialize all Stores for this DataContext.
+
+        Stores are a good fit for reading/writing objects that:
+            1. follow a clear key-value pattern, and
+            2. are usually edited programmatically, using the Context
+
+        Note that stores do NOT manage plugins.
+        """
+        for store_name, store_config in store_configs.items():
+            self._build_store_from_config(store_name, store_config)
+
+    # class method
+    @classmethod
+    def _get_global_config_value(
+        cls,
+        environment_variable: Optional[str] = None,
+        conf_file_section=None,
+        conf_file_option=None,
+    ) -> Optional[str]:
+        """
+        Returns global config value from environment variable or None
+        Args:
+            environment_variable (str): variable to return
+        Returns:
+            value of env variable or None
+        """
+        assert (conf_file_section and conf_file_option) or (
+            not conf_file_section and not conf_file_option
+        ), "Must pass both 'conf_file_section' and 'conf_file_option' or neither."
+        if environment_variable and os.environ.get(environment_variable, False):
+            return os.environ.get(environment_variable)
+        if conf_file_section and conf_file_option:
+            for config_path in AbstractDataContext.GLOBAL_CONFIG_PATHS:
+                config = configparser.ConfigParser()
+                config.read(config_path)
+                config_value = config.get(
+                    conf_file_section, conf_file_option, fallback=None
+                )
+                if config_value:
+                    return config_value
+        return None
+
+    def get_config_with_variables_substituted(self, config=None) -> DataContextConfig:
+        """
+        Takes DataContextConfig that is passed into original constructor and substitutes variables from os.environ.
+
+        *Note* Called every time the property `config_with_variables_substituted` is accessed.
+
+        Args:
+            config (DataContextConfig): Config with no substitutions (ie. passed into original constructor)
+        Returns:
+            DataContextConfig with variables substituted from os.environ
+
+        """
+        substituted_config_variables = substitute_all_config_variables(
+            self.config_variables,
+            dict(os.environ),
+            self.DOLLAR_SIGN_ESCAPE_STRING,
+        )
+        substitutions = {
+            **substituted_config_variables,
```

Comment:
```suggestion
```
I think you can get rid of this in the AbstractDataContext and only add config_variables in the BaseDataContext and FileDataContext